crates/biome_html_parser/src/lexer/mod.rs

```diff
@@ -233,6 +243,41 @@ impl<'src> HtmlLexer<'src> {
         }
     }
 
+    /// Consume an attribute value that is not quoted.
+    ///
+    /// See: https://html.spec.whatwg.org/#attributes-2 under "Unquoted attribute value syntax"
+    fn consume_unquoted_string_literal(&mut self) -> HtmlSyntaxKind {
+        let mut content_started = false;
+        let mut encountered_invalid = false;
+        while let Some(current) = self.current_byte() {
+            match current {
+                // these characters safely terminate an unquoted attribute value
+                b'\n' | b'\r' | b'\t' | b' ' | b'>' => break,
+                // these characters are absolutely invalid in an unquoted attribute value
+                b'?' | b'\'' | b'"' | b'=' | b'<' | b'`' => {
+                    encountered_invalid = true;
+                    break;
+                }
+                _ if current.is_ascii() => {
+                    self.advance(1);
+                    content_started = true;
+                }
+                _ => break,
+            }
+        }
+
+        if content_started && !encountered_invalid {
```

Comment:
Is it allowed for an unquoted attribute value to be empty? We might want to handle that case.

Reply:
No, that's a syntax error. If there is an equals sign, there must be a value that comes after it. An unquoted attribute value must have at least one character, otherwise its not present.